artifactory/commands/generic/setprops.go

```
@@ -31,16 +31,20 @@ func (setProps *SetPropsCommand) Run() error {
 		return err
 	}
 
-	resultItems, searchErr := searchItems(setProps.Spec(), servicesManager)
-
-	propsParams := GetPropsParams(resultItems, setProps.props)
+	reader, searchErr := searchItems(setProps.Spec(), servicesManager)
+	defer reader.Close()
+	propsParams := GetPropsParams(reader, setProps.props)
 	success, err := servicesManager.SetProps(propsParams)
 
 	result := setProps.Result()
 	result.SetSuccessCount(success)
-	result.SetFailCount(len(resultItems) - success)
+	totalLength, totalLengthErr := reader.Length()
```

This error is not checked.